### DISCUSSION TITLE
Ability to Skip Iteration or Use NaN

### DISCUSSION THREAD

**Discussion by Roaming-Mars** [snl-dakota/dakota#31] [Q&A]
I am performing a multidimensional parameter study and would like to allow my interface to fail. Does Dakota allow for the ability to skip an iteration/model if the interface script fails or, if not, is it possible to use NaN values if Dakota is expecting a real number? Thank you for your help!

**Comment by jadamstephens** [comment]
Yes. Your driver can write NaN for each expected response to the results file. Dakota will continue on.

Note that this won't work for all methods. Some methods, such as gradient-based optimizers, are not fault-tolerant in this way.

**Reply by Roaming-Mars** [reply]
That is great to hear! Is there any documentation on how Dakota accepts NaN? Specifically, my setup is expecting a float and throws an error when I try to put a NaN string through. Thank you so much!

**Reply by jadamstephens** [reply]
What's the error?

**Reply by Roaming-Mars** [reply]
The error is "TypeError: must be real number, not str"..."Item "--------------------------------------------------------------------------------" found while reading function values is not a valid floating point number."

**Reply by Roaming-Mars** [reply]
Also, does Dakota ignore nan in the statistical analysis? Thank you for your help!

**Reply by jadamstephens** [reply]
That looks like a Python exception. Is your driver written in Python? Maybe the driver is trying to write formatted output and is expecting a floating point number?

The second error is from Dakota. Somewhere or other, a long string of dashes is being written to your results file.
